### DISCUSSION TITLE
Show field type underneath field name in data model view and show all the time, not just when hovered over

### DISCUSSION THREAD

**Discussion by sxsws** [directus/directus#14027] [Feature Requests]
![Screenshot from 2022-06-21 15-47-20](https://user-images.githubusercontent.com/95102939/174830387-c99d9026-76eb-426e-80d2-396c8883fd5a.png)

Showing the field type underneath the field name will allow the type to be quickly seen  (instead of having to hover over each field) and by moving it underneath the field name will minimise the chance it being invisible. e.g. "Ma" in the image above. Also, I think it should be shown all the time, it is the 'data model' view, not the 'hover over elements to see the data model' view!

**Comment by benhaynes** [comment]
I agree! But I think I'd like to wait for the broader update/fix described here: https://github.com/directus/directus/discussions/9161#discussioncomment-2620819

That discussion/comment outlines splitting the data model's schema from the presentation and form layout. This will give the schema more space for showcasing this important information, without commingling the presentation side.

Thoughts on this direction?

![Settings-Data-Model-Collection](https://user-images.githubusercontent.com/522079/174836173-58348484-bc61-4b13-a67d-9a1e1beb2803.png)

**Reply by sxsws** [reply]
Nice! How do you quickly determine what table/collection a column is related to? e.g. I assume article_id is actually a relation to articles? The numbers on left make no sense!

This tabbed layout will also allow the easy addition for an 'Indexes'/'Indices' tab and I think 'Foreign Keys' should be a separate tab as there's lots of information to a fk!

Also, form layout looks great in the discussion! Please test the form layouts at different resolutions though because for higher resolutions, having the label on the left, a huge gap and then the input would be really poor readability. I think the admin user/s should be able to choose between 'label above/input below' or 'label left/input right'.

I'm also not a fan of using grey text for values. Readability is poor and "NULL" is an important concept in databases and I don't think there's any reason to grey it out!

**Reply by rijkvanzanten** [reply]
>  The numbers on left make no sense!

The numbers would be the database's [Ordinal Position](https://www.navicat.com/en/company/aboutus/blog/1693-all-about-ordinal_position) of the columns in the table 🙂

**Reply by benhaynes** [reply]
> I think the admin user/s should be able to choose between 'label above/input below' or 'label left/input right'.

This would be an option

> I'm also not a fan of using grey text for values. Readability is poor and "NULL" is an important concept in databases and I don't think there's any reason to grey it out!

Those are placeholders that show a default state when no value is present. Using gray text for placeholders is common, and we do it now. Also, this color is adjustable in the CSS overrides and upcoming Theme system.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
